components/PostImage/PostImage.js

```diff
@@ -2,15 +2,50 @@ import styles from './PostImage.module.css';
 import cx from 'classnames';
 import React from 'react';
 
+const useElementOnScreen = (options) => {
+  const imgRef = React.useRef(null);
+  const [isVisible, setIsVisible] = React.useState(false);
+  React.useEffect(() => {
+    const observer = new IntersectionObserver(([entry]) => {
+      setIsVisible(entry.isIntersecting);
+    }, options);
+    if (imgRef.current) {
+      observer.observe(imgRef.current);
+    }
+    return () => {
+      if (imgRef.current && isVisible) {
+        observer.disconnect(imgRef.current);
+      }
+    };
+  }, [imgRef, options]);
+
+  return [imgRef, isVisible];
+};
+
 export function PostImage({
   imageURL,
   fraction,
   aspectRatio,
   isLoading,
   setIsLoading,
   onDoubleClick,
+  tags,
+  postDimensions,
+  isActive,
 }) {
   const [isImgDoubleClicked, setIsImgDoubleCLicked] = React.useState(false);
+  const [isImgClicked, setIsImgCLicked] = React.useState(false);
+  const [imgRef, isVisible] = useElementOnScreen({
+    root: null,
+    rootMargin: '0px',
+    threshold: 1,
```

Comment:
This should be 0
```suggestion
    threshold: 0,
```

1 means that you want the entire element visible to mark at as visible. 0.5 means even if only 50% of it is visible, you should mark it as visible. While 0 means if it's visible at all (even 1px of it), mark it as visible. This would fix a bug you have, to see the bug:

1. View an image with tags, the icons will be visible.
2. Wait 3 seconds and the icon will go away.
3. Scroll just a little bit such that the image is partially invisible.
4. Scroll back such that the entire image is visible, the icon will show again. This is because the image was marked as invisible when it's top part was hidden, then was marked as visible again when it reappeared.

Reply:
On it.